hack/make-rules/test-cmd-util.sh

```diff
@@ -1371,6 +1351,125 @@ __EOF__
   # Make sure it's gone
   kube::test::get_object_assert bars "{{range.items}}{{$id_field}}:{{end}}" ''
 
+  # Test that we can create single item via apply
+  kubectl "${kube_flags[@]}" apply -f hack/testdata/TPR/foo.yaml
+
+  # Test that we have create a foo named test
+  kube::test::get_object_assert foos "{{range.items}}{{$id_field}}:{{end}}" 'test:'
+
+  # Test that the field has the expected value
+  kube::test::get_object_assert foos/test '{{.someField}}' 'field1'
+
+  # Test that apply an empty patch doesn't change fields
+  kubectl "${kube_flags[@]}" apply -f hack/testdata/TPR/foo.yaml
+
+  # Test that the field has the same value
+  kube::test::get_object_assert foos/test '{{.someField}}' 'field1'
+
+  # Test that apply has updated the subfield
+  kube::test::get_object_assert foos/test '{{.nestedField.someSubfield}}' 'subfield1'
+
+  # Update a subfield and then apply the change
+  sed 's/someSubfield: subfield1/someSubfield: modifiedSubfield/g' hack/testdata/TPR/foo.yaml | kubectl "${kube_flags[@]}" apply -f -
+
+  # Test that apply has updated the subfield
+  kube::test::get_object_assert foos/test '{{.nestedField.someSubfield}}' 'modifiedSubfield'
+
+  # Test that the field has the expected value
+  kube::test::get_object_assert foos/test '{{.nestedField.otherSubfield}}' 'subfield2'
+
+  # Delete a field and then apply the change
+  sed '/otherSubfield: subfield2/d' hack/testdata/TPR/foo.yaml | kubectl "${kube_flags[@]}" apply -f -
```

Comment:
make a series of files (`foo-updated-subfield.yaml`, `foo-deleted-field.yaml`, etc)